### PR TITLE
Add User Profile Edit Functionality US 20 and 22 SM + ZH

### PR DESCRIPTION
**Pull request by steve-meyers**
- Changed new user form to use form_tags instead of form_for
- Modified User strong params to work with form_tag syntax
- Added functionality to update user profile
- Added functionality to block user from updating to existing email (with flash message)
- [x] All tests passing?

### CODE REVIEW

File: app/views/users/new.html.erb [hunk 1]
```diff
@@ -1,28 +1,28 @@
 <h1>New User Registration</h1>
```
Comment (zachholcomb) [inline app/views/users/new.html.erb:2]: So we changed form_for back to form_tag for the moment to get the functionality up and running, refactor in to a form_for once everything is working later on

File: app/controllers/users_controller.rb [hunk 2]
```diff
@@ -18,9 +18,25 @@ def show
     require_user
   end
 
+  def edit
+    @user = User.find(session[:user])
+  end
+
+  def update
```
Comment (zachholcomb) [inline app/controllers/users_controller.rb:25]: update is a really long method now, we should split it off into private helper methods

File: app/views/users/new.html.erb (hunk 1 above)
Comment (jrsewell400) [reply app/views/users/new.html.erb:2]: looks good!

File: spec/features/visitor/create_spec.rb [hunk 3]
```diff
@@ -25,7 +25,7 @@
     fill_in :Zip, with: "66666"
     fill_in :Email, with: "chunky_lover@example.com"
     fill_in :Password, with: "123password"
-    fill_in :user_password_confirmation, with: "123password"
```
Comment (zachholcomb) [inline spec/features/visitor/create_spec.rb:28]: i think user_password_confirmation is a form_for convention that wasn't working with form_tag

Comment (jrsewell400) [reply spec/features/visitor/create_spec.rb:28]: good catch

File: app/controllers/users_controller.rb (hunk 2 above)
Comment (jrsewell400) [reply app/controllers/users_controller.rb:25]: Added to the to do list

**Review by jrsewell400 — COMMENTED** [review]
Left comments, I think everything looks good.

**Review by elguapogordo — APPROVED** [review]

**Comment by jrsewell400** [comment]
Looks good, added to do list items.  merging!

**Pull request merged**